### PR TITLE
Update vendoring tool to check on matplotlib colormap

### PR DESCRIPTION
**Pull request by Carreau**
This can check individual files

I do not update individual files, as so far I have been unable to find exact matches for the files I have checked, and I do not want to mix updating the tool to check for vendor files, from updating the files themselves (so that we can revert updating the files if necessary).

### CODE REVIEW

File: tools/check_vendored_modules.py [hunk 1]
```diff
@@ -14,7 +14,38 @@
...
+        shutil.rmtree(repo_path)
+
+    check_output(
+        [
+            "git",
+            "clone",
+            '--depth',
+            '1',
+            '--branch',
+            tag,
```
Comment (Carreau) [inline tools/check_vendored_modules.py:29]: That will not work with an exact hash, but is faster when giving a tag or a branch as it clones only what is necessary.

**Review by Czaki — APPROVED** [review]

**Comment by kcpevey** [comment]
@Carreau you have an approval on this. Are you ready to merge?

**Comment by Carreau** [comment]
Rebased on main.

**Comment by psobolewskiPhD** [comment]
Gentle bump: @Czaki you want to pull the trigger on this or shall we seek another review?
(I'm not really getting what this is doing, so not gonna approve or merge.)

**Review by Czaki — APPROVED** [review]
start 24h clock

**Comment by Carreau** [comment]
> start 24h clock

Thanks, much appreciated !

**Pull request merged**